### PR TITLE
Bumping Gitlab plugin API version to v4

### PR DESCRIPTION
**Pull request by fjsanpedro**
On June 22th Gitlab will remove version 3 from their codebase (currently deprecated). This PR bumps the version used by the plugin.

### CODE REVIEW

File: src/sentry_plugins/gitlab/client.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def get_issue(self, repo, issue_id):
                 ),
                 params={
                     # XXX(dcramer): this is an undocumented API
-                    'iid': issue_id,
+                    'issue_iid': issue_id,
```
Comment (DouweM) [inline src/sentry_plugins/gitlab/client.py]: @fjsanpedro Why was this change needed?

File: src/sentry_plugins/gitlab/plugin.py [hunk 2]
```diff
@@ -107,7 +107,7 @@ def create_issue(self, request, group, form_data, **kwargs):
         except Exception as e:
             self.raise_error(e)
 
-        return response['iid']
+        return response['issue_iid']
```
Comment (DouweM) [inline src/sentry_plugins/gitlab/plugin.py]: And this change? Per https://docs.gitlab.com/ee/api/issues.html#single-issue, GitLab still just uses `iid`.

Comment (fjsanpedro) [reply src/sentry_plugins/gitlab/plugin.py]: Solved

**Comment by DouweM** [comment]
I think that’s the path segment, not an actual separate query parameter.

El día 22 de mayo de 2018 a las 18:11:48, Francisco Javier López (
notifications@github.com) escrito:

File: src/sentry_plugins/gitlab/client.py [hunk 3]
```diff
@@ -45,7 +45,7 @@ def get_issue(self, repo, issue_id):
                 ),
                 params={
                     # XXX(dcramer): this is an undocumented API
-                    'iid': issue_id,
+                    'iids[]': issue_id,
```
Comment (DouweM) [inline src/sentry_plugins/gitlab/client.py]: Could we request `/projects/:repo/issues/:iid` here, instead of `/projects/:repo/issues?iids[]=:iid`?

I think that'll also allow us to drop `[0]` below, as well as the `try`/`except IndexError` stuff.

**Comment by fjsanpedro** [comment]
@DouweM PR updated

**Comment by DouweM** [comment]
@dcramer Could you please have a look at this PR? API v3 will be removed in GitLab 11.0 on June 22nd, and we wouldn't want the Sentry integration to break when that happens :)

**Comment by dcramer** [comment]
This lgtm but I'm handing off to @getsentry/app as I literally don't write code anymore :(

**Comment by stanhu** [comment]
Could we merge this please? :)

**Comment by dcramer** [comment]
Thanks!

**Comment by fjsanpedro** [comment]
Thanks @dcramer
